src/data_hub/plotting.py

```diff
@@ -19,43 +24,75 @@ def scatter(
     y: Optional[str] = None,
     size: Optional[str] = None,
 ):
-    """function to make a scatter plot from a Deep Origin dataframe, with support for interactivity"""
+    """function to make a scatter plot from a Deep Origin dataframe, with support for interactivity
+
+    Args:
+        df (DataFrame): Deep Origin dataframe
+        x (Optional[str], optional): name of column to use for x axis. Defaults to None.
+        y (Optional[str], optional): name of column to use for y axis. Defaults to None.
+        size (Optional[str], optional): name of column to use for size. Defaults to None.
+
+    `df` should be a Deep Origin dataframe with at least two numeric columns.
+
+    """
 
     figure_width = 500
     select_width = int(figure_width * 0.3)
-
-    if df.shape[1] < 2:
-        raise ValueError("DataFrame must contain at least two columns.")
+    js_code = _read_js_code()
 
     cols = df.attrs["metadata"]["cols"]
-    cols = [col["name"] for col in cols if col["type"] in ["float", "integer"]]
+    numeric_cols = [col["name"] for col in cols if col["type"] in ["float", "integer"]]
```

Comment:
Really feels like `["float", "integer"]` should be pulled out to a top-level def.

Reply:
will fix in next PR!